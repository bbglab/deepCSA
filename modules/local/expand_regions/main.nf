process EXPAND_REGIONS {

    tag "$meta.id"
    label 'process_high'

    container "docker.io/bbglab/deepcsa-core:0.0.2-alpha"

    input:
    tuple val(meta), path(panel)
    path (domains, optional: true)
    path (exons, optional: true)
    path (custom, optional: true)

    output:
    tuple val(meta), path("*with_hotspots.tsv") , emit: panel_increased
    tuple val(meta), path("hotspot_names.json") , emit: new_regions_json
    path "versions.yml"                         , topic: versions


    script:
    // def expansion = task.ext.expansion ?: 0
    def autoexons = params.omega_autoexons ? "--autoexons ${exons}" : ""
    def autodomains = params.omega_autodomains ? "--autodomains ${domains}" : ""
    def custom_regions = params.omega_subgenic_bedfile ? "--custom ${custom}" : ""
    """
    add_hotspots.py --panel_file ${panel} \\
                        ${autoexons} \\
                        ${autodomains} \\
                        ${custom_regions}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    """
    touch panel.with_hotspots.tsv
    touch hotspot_names.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
