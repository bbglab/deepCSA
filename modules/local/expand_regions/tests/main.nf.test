nextflow_process {
    name "Test EXPAND_REGIONS process"
    script "modules/local/expand_regions/main.nf"
    process "EXPAND_REGIONS"

    test("Should run without autoexons and autodomains") {
        when {
            params {
                omega_autoexons = false
                omega_autodomains = false
                omega_subgenic_bedfile = false
            }
            process {
                """
                input[0] = tuple(
                    [ id:'test', single_end:false ], 
                    file("${projectDir}/test_data/modules/consensus.exons_splice_sites.PPM1D.tsv")
                )
                input[1] = file("${projectDir}/test_data/modules/PPM1D_domains.bed4.bed")
                input[2] = file("${projectDir}/test_data/modules/PPM1D_exons.bed4.bed")
                input[3] = file("${projectDir}/test_data/dummy_file.tsv")
                """
            }
        }

        then {
            assert !process.success
        }
    }

    // test("Should run with autoexons and autodomains") {
    //     when {
    //         params {
    //             omega_autoexons = true
    //             omega_autodomains = true
    //             omega_subgenic_bedfile = false
    //         }
    //         process {
    //             """
    //             input[0] = tuple(
    //                 [ id:'test2', single_end:false ], 
    //                 file("${projectDir}/test_data/modules/consensus.exons_splice_sites.PPM1D.tsv")
    //             )
    //             input[1] = file("${projectDir}/test_data/modules/PPM1D_domains.bed4.bed")
    //             input[2] = file("${projectDir}/test_data/modules/PPM1D_exons.bed4.bed")
    //             input[3] = file("${projectDir}/test_data/dummy_file.tsv")
    //             """
    //         }
    //     }

    //     then {
    //         assert process.success
    //         assert snapshot(process.out).match()
    //         assert path(process.out.panel_increased.get(0)[1]).exists()
    //         assert path(process.out.new_regions_json.get(0)[1]).exists()
    //         assert path(process.out.versions).exists()
    //     }
    // }

    // test("Should run in stub mode") {
    //     when {
    //         process {
    //             """
    //             input[0] = tuple(
    //                 [ id:'test_stub', single_end:false ], 
    //                 file("${projectDir}/test_data/modules/consensus.exons_splice_sites.PPM1D.tsv")
    //             )
    //             input[1] = file("${projectDir}/test_data/modules/PPM1D_domains.bed4.bed")
    //             input[2] = file("${projectDir}/test_data/modules/PPM1D_exons.bed4.bed")
    //             input[3] = file("${projectDir}/test_data/dummy_file.tsv")
    //             """
    //         }
    //     }

    //     then {
    //         config {
    //             stub = true
    //         }
    //         assert process.success
    //         assert snapshot(process.out).match()
    //     }
    // }
}
 