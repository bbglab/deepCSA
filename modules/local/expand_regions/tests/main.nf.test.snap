{
    "Should run with autoexons and autodomains": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "exons_consensus_panel_with_hotspots.tsv:md5,d082d4b1355cbc3097aebd4eab3917b0"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "hotspot_names.json:md5,26890d8cc71b571e35ca4e49c51b6567"
                    ]
                ],
                "2": [
                    "versions.yml:md5,2a87f11ee825beb104e6cb60c40a4340"
                ],
                "new_regions_json": [
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "hotspot_names.json:md5,26890d8cc71b571e35ca4e49c51b6567"
                    ]
                ],
                "panel_increased": [
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "exons_consensus_panel_with_hotspots.tsv:md5,d082d4b1355cbc3097aebd4eab3917b0"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-19T09:08:34.567483549"
    }
}